vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helper/PresentationFactory.kt

```diff
@@ -0,0 +1,194 @@
+    ): KmmResult<Holder.PresentationResponseParameters> = catching {
+        params.parameters.verifyResponseType(presentationDefinition)
+        if (params.parameters.nonce == null) {
+            Napier.w("nonce is null in ${params.parameters}")
+            throw OAuth2Exception(Errors.INVALID_REQUEST)
+        }
+        val credentialSubmissions = inputDescriptorSubmissions
+            ?: holder.matchInputDescriptorsAgainstCredentialStore(
+                inputDescriptors = presentationDefinition.inputDescriptors,
+                fallbackFormatHolder = presentationDefinition.formats ?: clientMetadata?.vpFormats,
+            ).getOrThrow().toDefaultSubmission()
+
+        presentationDefinition.validateSubmission(
+            holder = holder,
+            clientMetadata = clientMetadata,
+            credentialSubmissions = credentialSubmissions
+        )
+
+        holder.createPresentation(
+            challenge = params.parameters.nonce,
+            audienceId = audience,
+            presentationDefinitionId = presentationDefinition.id,
+            presentationSubmissionSelection = credentialSubmissions,
+        ).getOrElse {
+            Napier.w("Could not create presentation", it)
+            throw OAuth2Exception(Errors.USER_CANCELLED)
+        }.also { container ->
+            clientMetadata?.vpFormats?.let { supportedFormats ->
+                container.verifyFormatSupport(supportedFormats)
+            }
+        }
+    }
+
+
+    suspend fun createSignedIdToken(
+        clock: Clock,
+        agentPublicKey: CryptoPublicKey,
+        parameters: AuthenticationRequestParametersFrom<*>,
+    ): KmmResult<JwsSigned?> = catching {
+        if (parameters.parameters.responseType?.contains(ID_TOKEN) != true) {
+            return@catching null
+        }
+        if (parameters.parameters.nonce == null) {
+            Napier.w("nonce is null in ${parameters.parameters}")
+            throw OAuth2Exception(Errors.INVALID_REQUEST)
+        }
+        val now = clock.now()
+        // we'll assume jwk-thumbprint
+        val agentJsonWebKey = agentPublicKey.toJsonWebKey()
+        val idToken = IdToken(
+            issuer = agentJsonWebKey.jwkThumbprint,
+            subject = agentJsonWebKey.jwkThumbprint,
+            subjectJwk = agentJsonWebKey,
+            audience = parameters.parameters.redirectUrl ?: parameters.parameters.clientId
+            ?: agentJsonWebKey.jwkThumbprint,
+            issuedAt = now,
+            expiration = now + 60.seconds,
+            nonce = parameters.parameters.nonce,
+        )
+        val jwsPayload = idToken.serialize().encodeToByteArray()
+        val signedIdToken =
+            jwsService.createSignedJwsAddingParams(payload = jwsPayload, addX5c = false).getOrElse {
+                Napier.w("Could not sign id_token", it)
+                throw OAuth2Exception(Errors.USER_CANCELLED)
+            }
+        return@catching signedIdToken
```

Comment:
I think the return is not necessary here

Reply:
fixed in f2a300dabe9855916935f35daf1408b409f71cc8